File: frontend/src/shared/api/history.ts
import { ClearHistory, DeleteHistoryEntry, ListHistory } from '../../../wailsjs/go/wails/HistoryService'

export interface HistoryEntry {
  id: number
  hostId?: string
  command: string
  executedAt: number
}

/** "" = every session, "local" = local shells only, otherwise a host ID. */
export type HistoryScope = '' | 'local' | string

export interface HistoryQuery {
  hostId?: HistoryScope
  search?: string
  limit?: number
  offset?: number
}

export async function listHistory(query: HistoryQuery = {}): Promise<HistoryEntry[]> {
  return ListHistory({
    hostId: query.hostId ?? '',
    search: query.search ?? '',
    limit: query.limit ?? 0,
    offset: query.offset ?? 0,
  })
}

export async function deleteHistoryEntry(id: number): Promise<void> {
  await DeleteHistoryEntry(id)
}

export async function clearHistory(scope: HistoryScope = ''): Promise<void> {
  await ClearHistory(scope)
}

File: internal/core/domain/history.go
package domain

// HistoryEntry is one captured command line. HostID is nil for a local
// shell session, or set to the originating host's ID for SSH.
type HistoryEntry struct {
	ID         int64
	HostID     *string
	Command    string
	ExecutedAt int64 // Unix seconds
}

// HistoryQuery filters HistoryRepository.List. HostID nil means "every
// session, local and SSH alike"; Limit <= 0 means "no limit".
type HistoryQuery struct {
	HostID *string
	Search string
	Limit  int
	Offset int
}
